src/components/SearchBar.jsx

```diff
@@ -7,13 +7,68 @@ const INTENT_VERB = 'OPEN'
 const INTENT_DOCTYPE = 'io.cozy.suggestions'
 const SUGGESTIONS_PER_SOURCE = 10
 
+const normalizeString = str => str.toString().toLowerCase().replace(/\//g, ' ').normalize('NFD').replace(/[\u0300-\u036f]/g, '').split(' ')
+
+const highlightQueryTerms = (searchResult, query) => {
+  const normalizedQueryTerms = normalizeString(query)
+  const normalizedResultTerms = normalizeString(searchResult)
+
+  const matchedIntervals = []
+  const spacerLength = 1
+  let currentIndex = 0
+
+  normalizedResultTerms.forEach(resultTerm => {
+    normalizedQueryTerms.forEach(queryTerm => {
+      const index = resultTerm.indexOf(queryTerm)
+      if (index >= 0) {
+        matchedIntervals.push({
+          from: currentIndex + index,
+          to: currentIndex + index + queryTerm.length
+        })
+      }
+    })
+
+    currentIndex += resultTerm.length + spacerLength
+  })
+
+  // matchedIntervals can overlap, so we merge them.
+  // - sort the intervals by starting index
+  // - add the first interval to the stack
+  // - for every interval,
+  // - - add it to the stck if it doesn't overlap with the stack top
+  // - - or extend the stack top if the start overlaps and the new interval's top is bigger
+  const mergedIntervals = matchedIntervals.sort((intervalA, intervalB) => intervalA.from > intervalB.from).reduce((computedIntervals, newInterval) => {
```

Comment:
I guess the algorithme is not mastered as it needs some comments.

Reply:
I found the algorithm not so trivial, enough to warrant a quick explanation in plain english.